### PR TITLE
FE: Monitoring Policies - unit tests for MonitoringPolicyBasicInforma…

### PR DESCRIPTION
**Pull request by faizanjanjua**
…tionEditAddForm

## Description
Add unit tests for MonitoringPolicyBasicInformationEditAddForm.vue

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2580

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Review by Shahbaz-dataq — COMMENTED** [review]
Approve

File: ui/tests/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.test.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  stubActions: false,
+  global: {
+    directives: {
+      focus: featherInputFocusDirective
+    },
+  }
+})
+
+describe('Test for MonitoringPolicyBasicInformationEditAddForm', () => {
+  beforeEach(()=>{
```
Comment (synqotik) [inline ui/tests/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.test.ts:30]: Should be `beforeEach(() => {` (spaces around arrow)

File: ui/tests/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.test.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    await wrapper.vm.$nextTick();
+    const policyNameInput = wrapper.findComponent('[data-test="policy-name-input"]');
+    expect(policyNameInput.exists()).toBe(true);
+    expect(policyNameInput.vm.modelValue).toBe(mockMonitoringPolicy.name); // Verify initial val
+  })
+  test('Policy description textarea reflects selected policy memo', async () => {
+    let store = useMonitoringPoliciesStore()
+    store.selectedPolicy = mockMonitoringPolicy 
+    await wrapper.vm.$nextTick();
+    const policyDesriptionInput = wrapper.findComponent('[data-test="policy-Description"]');
```
Comment (synqotik) [inline ui/tests/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.test.ts:53]: `policyDesriptionInput` -> `policyDescriptionInput`

File: ui/tests/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.test.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    expect(policyDesriptionInput.exists()).toBe(true);
+    expect(policyDesriptionInput.vm.modelValue).toBe(mockMonitoringPolicy.memo); // Verify initial val
+  })
+  test('ormattedTags computed property returns correctly formatted tags', async () => {
```
Comment (synqotik) [inline ui/tests/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.test.ts:57]: `formattedTags`

File: ui/tests/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.test.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    const CheckPolicyFormOnUndefined = wrapper.find('.policy-form')
+    expect(CheckPolicyFormOnUndefined).toBeTruthy()
+  })
+  test('Policy name input reflects selected policy name', async () => {
+    let store = useMonitoringPoliciesStore()
+    store.selectedPolicy = mockMonitoringPolicy 
+    await wrapper.vm.$nextTick();
+    const policyNameInput = wrapper.findComponent('[data-test="policy-name-input"]');
+    expect(policyNameInput.exists()).toBe(true);
+    expect(policyNameInput.vm.modelValue).toBe(mockMonitoringPolicy.name); // Verify initial val
```
Comment (synqotik) [inline ui/tests/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.test.ts:47]: Remove comment, here and elsewhere

File: ui/tests/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.test.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  global: {
+    directives: {
+      focus: featherInputFocusDirective
+    },
+  }
+})
+
+describe('Test for MonitoringPolicyBasicInformationEditAddForm', () => {
+  beforeEach(()=>{
+    let store = useMonitoringPoliciesStore()
```
Comment (synqotik) [inline ui/tests/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.test.ts:31]: If you use `beforeEach`, then you can add `let store` inside `describe` and then inside `beforeEach` can be `store = useMonitoringPoliciesStore()`. Then remove the `let store` inside all the `test` blocks.

**or else** remove the `beforeEach`, and have have `const store = useMonitoringPoliciesStore()` inside each `test` block.

**Review by synqotik — CHANGES_REQUESTED** [review]
Added a few comments.

Also, wait to merge this until I have merged LOK-2576. Then get latest from `develop`, you will probably need to make a few small changes:

- `Policy` has been replaced with `@/types/policies/MonitoringPolicy`
- there are some more required fields in that, so you'll need to add to `mockMonitoringPolicy`

**Review by synqotik — APPROVED** [review]

**Pull request merged**